sdk/src/program-manager.ts

```diff
@@ -39,6 +39,28 @@ import {
 import { logAndThrow } from "./utils.js";
 import { OwnedRecord } from "./models/record-provider/ownedRecord.js";
 
+/**
+ * Represents the options for deploying and upgrading a transaction in the Aleo network.
+ * This interface is used to specify the parameters required for building and submitting an deployment transaction.
+ *
+ * @property {string} program - The program source code to be deployed.
+ * @property {number} priorityFee - The optional priority fee to be paid for the transaction.
+ * @property {boolean} privateFee - If true, uses a private record to pay the fee; otherwise, uses the account's public credit balance.
+ * @property {RecordSearchParams | undefined} [recordSearchParams] - Optional parameters for searching for a record to pay the execution transaction fee.
+ * @property {KeySearchParams} [keySearchParams] - Optional parameters for finding the matching proving & verifying keys for the function.
+ * @property {string | RecordPlaintext | undefined} [feeRecord] - Optional fee record to use for the transaction.
+ * @property {PrivateKey} [privateKey] - Optional private key to use for the transaction.
+ */
+interface DeployOptions {
+    program: string;
+    priorityFee: number;
+    privateFee: boolean;
+    recordSearchParams?: RecordSearchParams;
+    keySearchParams?: KeySearchParams;
```

Comment:
You can't find existing keys for a deployment, it'll always need to synthesize the keys.

```suggestion
```